### DISCUSSION TITLE
Using "today" inside Query Template will work differently if we are using "SharePoint Search" vs "Microsoft Search"

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#3938] [Q&A]
I have a SharePoint Online site collection with Pacific Time zone (US & Canada), and i have a Date/Time SharePoint site column named DueDate, and i created a RefinableDate00 which is linked to the site column crawl property.

Now i use this Query Template to get the items which have the DueDate equals Today inside PnP Search Result web part:-

```
{searchTerms}
ContentType=Assets
(RefinableDate00=today )
```

I have noted a major differences how this Query template works when having SharePoint Search vs Microsoft Search, as follow:-

1) let say i added 2 list items and i define its DueDate as follow:-

a) First Item:- 24 August @ 11:30  pm. where this date/time will be stored inside SharePoint in UTC time zone, so the value will be 25 August @ 6:30 am.
b) Second Item:- 24 August @ 12:00 am. where this date/time will be stored inside SharePoint in UTC time zone, so the value will be 24 August @ 7 am

2) Now it is 24 August @ 2:30 pm Pacific Time +  9:30 pm UTC.
3) when i run the above Query Template using the Microsoft Search, i will only get second item
4) while when i run the above Query Template using SharePoint Search i will get the 2 list items.

so seems the SharePoint Search do respect the site time zone and get accurate results, since in my case the 2 items are on 24 August (based on the site time zone) and both items have their DueDate on 24 August based on the site time zone, while using Microsoft Search the search will be based on UTC and i will not get accurate results based on the site collection time zone. 

so can anyone advice why we got this 2 behavior ? is this an issue in Microsoft Search?

**Comment by kasperbolarsen** [comment]
SharePoint is using UTC internally when storing datetimes. The user interface will correct the UTC timestamps to local time, however the SharePoint Search API will always return the UTC timestamps.
Looking at the results returned by the Graph Explorer ( https://graph.microsoft.com/v1.0/search/query + entiryTypes: drivetiem) it sure looks like the datetimes are UTC
![image](https://github.com/user-attachments/assets/7353a38c-2e5c-48a1-a5e5-cdfa27e48816)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen Yes i know that sharepoint stores the dates in UTC for items and documents.. but my question is how using the keyword "today" inside Query Template works differently when using Microsoft search and when using SharePoint search as the source of the Search Result pnp web part.. SharePoint search will change the 'today' to local time (not sure if it is the user browser time or the site collection time) while Microsoft search will return the "today" based on UTC time zone... I already gave an example on this how i will get 2 items when using SharePoint search while i only got one item when using Microsoft Search

**Comment by kasperbolarsen** [comment]
According to my testing the queries in SharePoint and Microsoft Search are using the same value for {Today}
Any chance you can run a similar test and check the data in the network tab?
 
![image](https://github.com/user-attachments/assets/da47cc70-3c3f-4a9a-982b-fd023a22efd5)
![image](https://github.com/user-attachments/assets/60a55391-fbe6-4889-94ec-1568d3b7fde6)
